### PR TITLE
r/aws_wafv2: Add missing values to `text_transformation` argument

### LINKED ISSUE #20545: aws_wafv2_web_acl does not support all available text_transformation types
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

It seems `aws_wafv2_web_acl` `text_transformation` does not support all types from [Amazon docs](https://docs.aws.amazon.com/waf/latest/APIReference/API_TextTransformation.html). Terraform docs also only specify some of the possible values, [not all](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/wafv2_rule_group#text-transformation)

### New or Affected Resource(s)

* aws_wafv2_web_acl 

### Error

```hcl
 Error: expected rule.1.statement.0.sqli_match_statement.0.text_transformation.2.type to be one of [CMD_LINE COMPRESS_WHITE_SPACE HTML_ENTITY_DECODE LOWERCASE NONE URL_DECODE], got SQL_HEX_DECODE
│ 
│   with module.wafv2.aws_wafv2_web_acl.default[0],
│   on .terraform/modules/wafv2/rules.tf line 85, in resource "aws_wafv2_web_acl" "default":
│   85: resource "aws_wafv2_web_acl" "default" {
│ 
```

### Versions:
```
$ terraform -v
Terraform v1.0.4
on linux_amd64
+ provider registry.terraform.io/hashicorp/aws v3.53.0
+ provider registry.terraform.io/hashicorp/template v2.2.0
```
### References

* #0000

### PR DESCRIPTION
**Pull request by brunhil**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #20545. Add missing values to `text_transformation` argument for `aws_wafv2_web_acl` and `aws_wafv2_rule_group` resources

### CODE REVIEW

File: aws/wafv2_helper.go [hunk 1]
```diff
@@ -382,6 +382,21 @@ func wafv2TextTransformationSchema() *schema.Schema {
 						wafv2.TextTransformationTypeLowercase,
 						wafv2.TextTransformationTypeNone,
 						wafv2.TextTransformationTypeUrlDecode,
+						wafv2.TextTransformationTypeBase64Decode,
```
Comment (breathingdust) [inline aws/wafv2_helper.go:385]: As described in (#14601) the `aws-go-sdk` has added `_Values()` suffixed const arrays so we can actually replace this whole statement with:

```
validation.StringInSlice(wafv2.TextTransformationType_Values(), false),
```
This way the validation will automatically update as we update `aws-go-sdk` versions.

File: website/docs/r/wafv2_rule_group.html.markdown [hunk 2]
```diff
@@ -520,7 +520,9 @@ The `single_query_argument` block supports the following arguments:
 The `text_transformation` block supports the following arguments:
 
 * `priority` - (Required) The relative processing order for multiple transformations that are defined for a rule statement. AWS WAF processes all transformations, from lowest priority to highest, before inspecting the transformed content.
-* `type` - (Required) The transformation to apply, you can specify the following types: `NONE`, `COMPRESS_WHITE_SPACE`, `HTML_ENTITY_DECODE`, `LOWERCASE`, `CMD_LINE`, `URL_DECODE`. See the [documentation](https://docs.aws.amazon.com/waf/latest/APIReference/API_TextTransformation.html) for more details.
+* `type` - (Required) The transformation to apply, you can specify the following types: `NONE`, `COMPRESS_WHITE_SPACE`, `HTML_ENTITY_DECODE`, `LOWERCASE`, `CMD_LINE`, `URL_DECODE`, `BASE64_DECODE`, `HEX_DECODE`, `MD5`, `REPLACE_COMMENTS`,
+`ESCAPE_SEQ_DECODE`, `SQL_HEX_DECODE`, `CSS_DECODE`, `JS_DECODE`, `NORMALIZE_PATH`, `NORMALIZE_PATH_WIN`, `REMOVE_NULLS`, `REPLACE_NULLS`, `BASE64_DECODE_EXT`, `URL_DECODE_UNI`, `UTF8_TO_UNICODE`.
+See the [documentation](https://docs.aws.amazon.com/waf/latest/APIReference/API_TextTransformation.html) for more details.
```
Comment (breathingdust) [inline website/docs/r/wafv2_rule_group.html.markdown:525]: As we are now using on the `TextTransformation_Values()` function, we should link to the TextTransformation documentation for specific values.

```suggestion
* `type` - (Required) The transformation to apply, please refer to the Text Transformation [documentation](https://docs.aws.amazon.com/waf/latest/APIReference/API_TextTransformation.html) for more details.
```

**Review by breathingdust — CHANGES_REQUESTED** [review]
Hi @brunhil! Thanks for the contribution, just a couple of small changes and it LGTM!

File: aws/wafv2_helper.go (hunk 1 above)
Comment (brunhil) [reply aws/wafv2_helper.go:385]: Wow! Thanks for this tidbit of info, super slick. Will get this updated shortly!

**Comment by breathingdust** [comment]
LGTM 🚀 

Verified Acceptance Tests in us-west-2


```bash
make testacc TESTARGS='-run=TestAccAwsWafv2WebACL_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAwsWafv2WebACL_ -timeout 180m

--- PASS: TestAccAwsWafv2WebACL_minimal (32.46s)
--- PASS: TestAccAwsWafv2WebACL_disappears (36.71s)
--- PASS: TestAccAwsWafv2WebACL_Update_rule (70.53s)
--- PASS: TestAccAwsWafv2WebACL_RateBased_maxNested (70.73s)
--- PASS: TestAccAwsWafv2WebACL_IPSetReference_basic (81.35s)
--- PASS: TestAccAwsWafv2WebACL_tags (84.90s)
--- PASS: TestAccAwsWafv2WebACL_ManagedRuleGroup_basic (87.94s)
--- PASS: TestAccAwsWafv2WebACL_RateBased_forwardedIPConfig (90.28s)
--- PASS: TestAccAwsWafv2WebACL_RateBased_basic (95.37s)
--- PASS: TestAccAwsWafv2WebACL_Update_nameForceNew (99.56s)
--- PASS: TestAccAwsWafv2WebACL_GeoMatch_forwardedIPConfig (101.44s)
--- PASS: TestAccAwsWafv2WebACL_Custom_requestHandling (109.64s)
--- PASS: TestAccAwsWafv2WebACL_Custom_response (111.87s)
--- PASS: TestAccAwsWafv2WebACL_RuleGroupReference_basic (121.12s)
--- PASS: TestAccAwsWafv2WebACL_GeoMatch_basic (134.42s)
--- PASS: TestAccAwsWafv2WebACL_Operators_maxNested (137.16s)
--- PASS: TestAccAwsWafv2WebACL_Update_ruleProperties (138.10s)
--- PASS: TestAccAwsWafv2WebACL_basic (146.87s)
--- PASS: TestAccAwsWafv2WebACL_IPSetReference_forwardedIPConfig (152.73s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	156.462s

make testacc TESTARGS='-run=TestAccAwsWafv2RuleGroup_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAwsWafv2RuleGroup_ -timeout 180m
--- PASS: TestAccAwsWafv2RuleGroup_GeoMatchStatement (66.88s)
--- PASS: TestAccAwsWafv2RuleGroup_SizeConstraintStatement (72.83s)
--- PASS: TestAccAwsWafv2RuleGroup_GeoMatchStatement_ForwardedIPConfig (82.54s)
--- PASS: TestAccAwsWafv2RuleGroup_RuleAction (82.85s)
--- PASS: TestAccAwsWafv2RuleGroup_Disappears (19.80s)
--- PASS: TestAccAwsWafv2RuleGroup_ByteMatchStatement (90.30s)
--- PASS: TestAccAwsWafv2RuleGroup_RegexPatternSetReferenceStatement (91.64s)
--- PASS: TestAccAwsWafv2RuleGroup_RuleAction_CustomResponse (95.92s)
--- PASS: TestAccAwsWafv2RuleGroup_RuleAction_CustomRequestHandling (98.03s)
--- PASS: TestAccAwsWafv2RuleGroup_ChangeCapacityForceNew (98.13s)
--- PASS: TestAccAwsWafv2RuleGroup_basic (103.31s)
--- PASS: TestAccAwsWafv2RuleGroup_ChangeMetricNameForceNew (35.36s)
--- PASS: TestAccAwsWafv2RuleGroup_updateRuleProperties (110.51s)
--- PASS: TestAccAwsWafv2RuleGroup_updateRule (36.78s)
--- PASS: TestAccAwsWafv2RuleGroup_XssMatchStatement (121.14s)
--- PASS: TestAccAwsWafv2RuleGroup_IpSetReferenceStatement (125.86s)
--- PASS: TestAccAwsWafv2RuleGroup_LogicalRuleStatements (133.49s)
--- PASS: TestAccAwsWafv2RuleGroup_Tags (138.62s)
--- PASS: TestAccAwsWafv2RuleGroup_ChangeNameForceNew (156.07s)
--- PASS: TestAccAwsWafv2RuleGroup_IpSetReferenceStatement_IPSetForwardedIPConfig (177.77s)
--- PASS: TestAccAwsWafv2RuleGroup_ByteMatchStatement_FieldToMatch (187.36s)
--- PASS: TestAccAwsWafv2RuleGroup_SqliMatchStatement (231.49s)
--- PASS: TestAccAwsWafv2RuleGroup_Minimal (237.17s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	240.339s
```

**Pull request merged**